src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/ExporterClientValidation.cs

```diff
@@ -1,29 +0,0 @@
-// Copyright The OpenTelemetry Authors
-// SPDX-License-Identifier: Apache-2.0
-
-namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
-
-internal static class ExporterClientValidation
-{
-    internal static void EnsureUnencryptedSupportIsEnabled(OtlpExporterOptions options)
```

Comment:
@Kielek I think we should mention this in the CHANGELOG. Thinking something like...

* OtlpExporter will no longer throw an exception when the `System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport` `AppContext` switch is NOT set AND using `OtlpExportProtocol.Grpc` to send to an insecure ("http") endpoint. `System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport` is not required to be set [when using .NET 5 or newer](https://learn.microsoft.com/aspnet/core/grpc/troubleshoot?view=aspnetcore-8.0#call-insecure-grpc-services-with-net-core-client). ([#5486](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5486))

Reply:
The exception was thrown only on .NET Core 3.x. From thi method:

```csharp
        // This verification is only required for .NET Core 3.x
        if (version.Major != 3)
        {
            return;
        }
```

Do you really want to put this info?
If so, I would go with this:

```md
* OtlpExporter will no longer throw an exception on .NET Core 3.1
   when the `System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport`
  `AppContext` switch is NOT set AND using `OtlpExportProtocol.Grpc`
  to send to an insecure ("http") endpoint.
  `System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport`
  is not required to be set [when using .NET 5 or newer](https://learn.microsoft.com/aspnet/core/grpc/troubleshoot?view=aspnetcore-8.0#call-insecure-grpc-services-with-net-core-client).
  ([#5486](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5486))
```

Reply:
I don't feel passionately that we absolutely need it. Just thinking it may be a nice hint to users that they can go and remove `AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);` if they added it at one point or another due to the exception being thrown.

Reply:
275076dfbf3a62fa3f55fded7457da5799d6aee8